pkg/util/kube/predicates.go

```diff
@@ -191,15 +195,13 @@ func (r *vmForVMClassPredicate) getVM(
 
 	data, err := runtime.DefaultUnstructuredConverter.ToUnstructured(inObj)
 	if err != nil {
-		return nil, fmt.Errorf("failed to convert to unstructured: %w", err)
+		return nil, "", fmt.Errorf("failed to convert to unstructured: %w", err)
 	}
 	outObj := unstructured.Unstructured{Object: data}
 
-	if kind := outObj.GetKind(); kind != "VirtualMachine" {
-		return nil, fmt.Errorf("kind=%q, expected kind=VirtualMachine", kind)
-	}
+	kind := outObj.GetKind()
 
-	return &outObj, nil
+	return &outObj, kind, nil
 }
 
 func (r *vmForVMClassPredicate) getVMClass(
```

Comment:
Closest GH will let me comment: I don't believe the ordering here is quite right. We should check the namespace-scoped first as a I'm pretty sure that a NS-scoped request w/o a namespace will be defaulted to "default" NS.

Reply:
@akutz  please comment on @bryanv question.